nntrainer/tensor/tensor_wrap_specs.h

```diff
@@ -70,6 +74,40 @@ typedef std::tuple<TensorDim, Tensor::Initializer, bool, const std::string,
                    TensorLifespan>
   VarGradSpec;
 
+/**
+ * @brief Tensor Specification which describes how this tensor should be
+ * allocated and managed
+ *
+ */
+struct TensorSpecV2 {
+  enum class RequestType {
+    PLACEHOLDER,
+    UNIQUE,
+    READ_ONLY_VIEW,
```

Comment:
how about keeping just 1 view and 1 shared:
VIEW: non-modifiable internal memory structure (shared with source tensor) with size <= source tensor size
SHARED: modifiable internal memory structure (shared with source tensor) with size <= source tensor size

This keeps view semantics same as string_view (https://en.cppreference.com/w/cpp/string/basic_string_view).

Reply:
That is can't be done. we need three concept. 

1. view changing data (MAYBE_MODIFYING_VIEW)
2. view not changing data but changing meta (READ_ONLY_VIEW)
3. share everything (SHARED)

We can seperate this concept out to another dimension but it will be dirtier.

Reply:
I like the concept. Let me delay this until there seems a way to represent the third state.